### PR TITLE
Feature/k shingling

### PR DESCRIPTION
**Pull request by rmsamaral**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,17 +7,50 @@
...
-    <groupId>junit</groupId>
-    <artifactId>junit</artifactId>
-    <version>${junit.version}</version>
-    <scope>test</scope>
-  </dependency>
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
+      <version>3.5</version>
```
Comment (rmsamaral) [inline pom.xml]: create property for the version.

File: src/main/java/KShingler.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        System.out.println("inputSize:" + inputSize);
+        System.out.println("input: " + input);
+        System.out.println("Error: " + e.getMessage());
+      }
+    }
+
+    return new ArrayList<>(output);
+  }
+
+  private void validateInputString(String input) {
```
Comment (rmsamaral) [inline src/main/java/KShingler.java]: Create a separate class for the validations (KShinglerValidator)

File: src/main/java/Shingler.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import java.util.List;
+
+public interface Shingler {
+  List<String> shingle(int i, String input);
```
Comment (rmsamaral) [inline src/main/java/Shingler.java]: Change i to k.

File: src/test/java/StringTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class StringTestUtils {
```
Comment (rmsamaral) [inline src/test/java/StringTestUtils.java]: Create a unit test class for this utility to ensure the behaviour under edge cases.

File: src/test/resources/features/k_shingler.feature [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    When the value of K is 5
+    And the input string value is "  "
+    Then it must not throw java.lang.IllegalArgumentException
+    
+  Scenario Outline: "Happy path" scenarios for the singles with their expected values
+    When the value of K is <size_of_k>
+    And the input string value is <input_value>
+    Then the shingle set, as a CSV, will be <shingle_as_csv>
+    Examples:
+      | size_of_k | input_value       | shingle_as_csv                                                                            |
```
Comment (rmsamaral) [inline src/test/resources/features/k_shingler.feature]: - value_of_k instead of size_of_k
- expected_shingle_as_csv instead of shingle_as_csv
